### PR TITLE
Rle

### PR DESCRIPTION
**Pull request by AImani2**

### CODE REVIEW

File: src/main/java/imani/game/Grid.java [hunk 1]
```diff
@@ -27,6 +30,57 @@ public void enliven(int x, int y) {
...
+                if (matcher.find()) {
+                    x = 0;
+                    y = 0;
+                }
+            }
+
+            StringBuilder num = new StringBuilder();
+            for (char character : line.toCharArray()) {
+                if (Character.isDigit(character)) {
+                    num.append(character);
```
Comment (andrewoid) [inline src/main/java/imani/game/Grid.java:64]: This doesn't do what you think it does.

File: src/main/java/imani/game/GridFrame.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+                String rleData = new String(Files.readAllBytes(rleFile.toPath()));
+
+                Grid grid = new Grid(40,40);
+                grid.importRLE(rleData);
+
+                SwingUtilities.invokeLater(() -> {
+                    new GridFrame(grid).setVisible(true);
+                });
+            } catch (IOException e) {
+                System.err.println("Error reading RLE file: " + e.getMessage());
```
Comment (andrewoid) [inline src/main/java/imani/game/GridFrame.java]: e.printStackTrace()

File: src/main/java/imani/game/Grid.java (hunk 1 above)
Comment (AImani2) [reply src/main/java/imani/game/Grid.java:64]: I am going to take a look at this

Comment (AImani2) [reply src/main/java/imani/game/Grid.java:64]: how do i create a test for my importRle method?
is my given the file that i enter into the command line arguments?

Comment (andrewoid) [reply src/main/java/imani/game/Grid.java:64]: You can put whatever String you want into this to test it. It doesn't need to be from a File.

File: src/main/java/imani/game/Grid.java [hunk 3]
```diff
@@ -27,6 +30,60 @@ public void enliven(int x, int y) {
...
+            if (line.startsWith("#")) {
+                continue;
+            }
+
+            if (line.startsWith("x")) {
+                Pattern pattern = Pattern.compile("x = (\\d+), y = (\\d+), rule = (.+)");
+                Matcher matcher = pattern.matcher(line);
+                if (matcher.find()) {
+                    x = 0;
+                    y = 0;
```
Comment (andrewoid) [inline src/main/java/imani/game/Grid.java:56]: Why are you setting this to 0?

File: src/main/java/imani/game/Grid.java [hunk 4]
```diff
@@ -27,6 +30,60 @@ public void enliven(int x, int y) {
...
+
+                    switch (character) {
+                        default:
+                            throw new IllegalArgumentException("Unexpected character: " + character);
+                        case 'b':
+                            x += count;
+                            break;
+                        case 'o':
+                            for (int i = 0; i < count; i++) {
+                                if (x < gameBoard[0].length && y < gameBoard.length) {
```
Comment (andrewoid) [inline src/main/java/imani/game/Grid.java:77]: You need to make sure the size of the board changes when you load an RLE

File: src/main/java/imani/game/GridComponent.java [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        addMouseListener(new MouseListener() {
+            @Override
+            public void mouseClicked(MouseEvent e) {
+
+                int x = e.getX() / cellSize;
+                int y = e.getY() / cellSize;
+
+
+                if (x < grid.getGameBoard().length && y < grid.getGameBoard()[0].length) {
+                    grid.enliven(x, y);
```
Comment (andrewoid) [inline src/main/java/imani/game/GridComponent.java]: If the cell is already on, you should turn it off.

File: src/main/java/imani/game/GridFrame.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+public class GridFrame extends JFrame {
+
+    private final GridComponent gridComponent;
+
+    public GridFrame(Grid grid) {
+
+        setTitle("Conway's Game of Life");
+
+        //Grid grid = new Grid(40, 40);
```
Comment (andrewoid) [inline src/main/java/imani/game/GridFrame.java]: delete this line.

File: src/main/java/imani/game/GridFrame.java [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+                grid.importRle(rleData);
+
+                SwingUtilities.invokeLater(() -> {
+                    new GridFrame(grid).setVisible(true);
+                });
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        } else {
+            System.out.println("Usage: java imani.game.GridFrame <path_to_rle_file>");
```
Comment (andrewoid) [inline src/main/java/imani/game/GridFrame.java]: If there are no command line arguments then open it up to an empty grid.

File: src/main/java/imani/game/Grid.java (hunk 1 above)
Comment (AImani2) [reply src/main/java/imani/game/Grid.java:64]: Thank you.
Professor, my program worked before. It ran, and I have a screenshot of it working. I am not sure why but now it has an error. I can't seem to figure it out. 
RLEFileReader
RLEFileReader
Would you be able to take a look at it and see what is wrong?

File: src/main/java/imani/game/GridFrame.java (hunk 7 above)
Comment (AImani2) [reply src/main/java/imani/game/GridFrame.java]: working on your comments

File: src/main/java/imani/game/Grid.java (hunk 3 above)
Comment (AImani2) [reply src/main/java/imani/game/Grid.java:56]: the x and y are the coordinates of the grid, my program colors the grid from the top left corner

**Review by andrewoid — APPROVED** [review]

**Pull request merged**